backend/main_test.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+const testUploadDir = "test_uploads"
+
+func TestMain(m *testing.M) {
+	// Ensure test uploads directory is cleaned up before and after tests
+	os.RemoveAll(testUploadDir)
+	os.Mkdir(testUploadDir, 0755)
+	code := m.Run()
+	os.RemoveAll(testUploadDir)
+	os.Exit(code)
+}
+
+func TestUploadHandlerContentLengthExceedsLimit(t *testing.T) {
+	reqBody := bytes.NewBuffer(make([]byte, maxUploadSize+1))
+	req := httptest.NewRequest(http.MethodPost, "/upload", reqBody)
+	req.Header.Set("Content-Length", "104857601")
```

Comment:
why hardcode ?
why not use the same (maxUploadSize+1) used while creating reqBody